cql3/restrictions/statement_restrictions.hh

```diff
@@ -445,7 +455,17 @@ public:
     }
 
     bool ck_restrictions_need_filtering() const {
-        return _partition_key_restrictions->has_unrestricted_components(*_schema) || _clustering_columns_restrictions->needs_filtering(*_schema);
+        if (_clustering_columns_restrictions->empty()) {
+            return false;
+        }
+
+        return _partition_key_restrictions->has_unrestricted_components(*_schema)
+        || _clustering_columns_restrictions->needs_filtering(*_schema)
+        // This function is called after need_filtering() returns true.
```

Comment:
If so, that should be mentioned prominently in this function's blurb, don't you think?  Is it dangerous or wrong to call this function _before_ need_filtering() returned true?

Reply:
This is not required, that was just to help understand what's going on.
I rewrote the comment completely, should be clearer now.